### PR TITLE
Cpp check - use cmake's built in cpp check property

### PR DESCRIPTION
**Pull request by r1viollet**
# Description

As per morrisonlevi's proposal, avoid using manual lookup of files in favour of the cmake cpp check support.

# Motivation

A more generic usage of cpp check.

# Additional notes

This makes it slightly slower to execute cpp check as it considers all Zend files.

### CODE REVIEW

File: cmake/Modules/CppCheck.cmake [hunk 1]
```diff
@@ -1,31 +1,21 @@
 # Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 # This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022-Present Datadog, Inc.
 
+set(CMAKE_VERBOSE_MAKEFILE on)
```
Comment (r1viollet) [inline cmake/Modules/CppCheck.cmake:4]: to be removed (for debug only)

File: cmake/Modules/CppCheck.cmake [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
-      # --check-config #check what header files are missing
-      ${CPPCHECK_ALL_SOURCE_FILES}
+      "--force"
+      "--error-exitcode=1"
       )
+    # Let user define his own cpp check commands if needed
+    if(NOT DEFINED CACHE{CMAKE_C_CPPCHECK})
+        set(CMAKE_CXX_CPPCHECK "${CPP_CHECK_COMMAND};--std=c++11")
+        set(CMAKE_C_CPPCHECK "${CPP_CHECK_COMMAND};--std=c11")
+    endif()
```
Comment (morrisonlevi) [inline cmake/Modules/CppCheck.cmake:26]: Each language should have its own conditional, I think? It's possible a user sets one but not the other.

**Review by morrisonlevi — COMMENTED** [review]
I like this direction, but have some questions/suggestions.

File: cmake/Modules/CppCheck.cmake (hunk 2 above)
Comment (r1viollet) [reply cmake/Modules/CppCheck.cmake:26]: true

File: CppCheckSuppressions.txt [hunk 3]
```diff
@@ -1,5 +1,13 @@
-// To exclude php include warnings (if include libraries are added)
-// *:*/php/.*-nts-ndebug*
-// toomanyconfigs:*
+// To exclude php include warnings from Zend
```
Comment (r1viollet) [inline CppCheckSuppressions.txt:1]: reminder to remove this, if the new strategy works.

File: CppCheckSuppressions.txt [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
+duplicateExpression:*/php/*-nts-ndebug*
+variableScope:*/php/*-nts-ndebug*
+unusedLabelConfiguration:*/php/*-nts-ndebug*
+unreadVariable:*/php/*-nts-ndebug*
+unusedLabelConfiguration:*/php/*-nts-ndebug*
+unsignedLessThanZero:*/php/*-nts-ndebug*
+ConfigurationNotChecked:*/php/*-nts-ndebug*
```
[the remark this answers is not in the document]
Comment (r1viollet) [reply CppCheckSuppressions.txt]: Yes, new version takes this request into account.

File: cmake/Modules/CppCheck.cmake [hunk 5]
```diff
@@ -1,31 +1,38 @@
 # Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 # This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022-Present Datadog, Inc.
-
+option(DD_STATIC_ANALYSIS "Enable static analysis tooling" OFF)
```
[the remark this answers is not in the document]
Comment (r1viollet) [reply cmake/Modules/CppCheck.cmake]: Fixed and propagated downstream

**Review by morrisonlevi — APPROVED** [review]
Looks good to me! I'm a bit suspicious though: are there really no warnings outside of PHP? I would have expected to have to fix at least _something_, you know?

**Comment by r1viollet** [comment]
> Looks good to me! I'm a bit suspicious though: are there really no warnings outside of PHP? I would have expected to have to fix at least _something_, you know?

The only warning I ran into was fixed in an earlier PR and related to scope of a variable.

**Pull request merged**